src/cpp/src/image_generation/models/clip_text_model.cpp

```diff
@@ -124,12 +124,25 @@ ov::Tensor CLIPTextModel::infer(const std::string& pos_prompt, const std::string
         }
     };
 
+    ov::PartialShape compiled_input_partial_shape =
+        m_request.get_compiled_model().get_runtime_model()->input(0).get_partial_shape();
+
     ov::Tensor input_ids = m_request.get_input_tensor();
-    input_ids.set_shape({text_embedding_batch_size, m_config.max_position_embeddings});
+
+    if (compiled_input_partial_shape.is_dynamic()) {
+        input_ids.set_shape({text_embedding_batch_size, m_config.max_position_embeddings});
+    } else {
+        auto compiled_input_shape = input_ids.get_shape();
+        OPENVINO_ASSERT(compiled_input_shape.size() == 2, "CLIP text encoder model input must have rank of 2");
+        OPENVINO_ASSERT(text_embedding_batch_size <= compiled_input_shape[0],
+                        "text_embedding_batch_size > CLIP text encoder model batch size");
+        OPENVINO_ASSERT(m_config.max_position_embeddings == compiled_input_shape[1],
+                        "max_position_embeddings != what CLIP text encoder model was compiled for.");
```

Comment:
can we print in these asserts that are actual numbers?
E.g. `max_position_embeddings (77) != what CLIP text encoder model was compiled for (55).` is more informative than `max_position_embeddings != what CLIP text encoder model was compiled for.`